### DISCUSSION TITLE
Improvement suggestions for lxconsole discussion

### DISCUSSION THREAD

**Discussion by matthewalanpenning** [PenningLabs/lxconsole#1] [Announcements]
If you have any suggestions on improving lxconsole, please leave your feedback here.

**Comment by ghost** [comment]
Hey Matthew. 

I suggest to buld in OpenID Authentication SSO so LXConsole could seamlessly be integrated into an existing environment.

I am a webdesigner so I could not contribute in this case....

Thanks for building the INCUS support :-) 

Cheers, 
Magnus.

**Comment by skea999** [comment]
hi, i upvoted the comment for the openid, it is really usefull. By the way i know some tricks, you can set a reverse proxy to serve this application and you can authenticate with openid on the reverse proxy, it will be usefull to map rules inside the application, like admins can do allthing, restricted users can only do some basic stuffs and something like that, also it will be usefull to assign some users or groups to projects.

The second thing that i want to discuss is the possibility to move instances between hosts not in a cluster, i can't help here, i don't know if it is possible to do such a thing without downtimes, but sure it is usefull to move instances between hosts not in a cluster also with downtimes, this can be simple (i guess), it is like: take the backup, move the backup to the other host, restore the backup.

The third thing is making the application scalable, this is essential for todays standard, since this application is in fact an hypervisor capable to do allthings, it need to be runned in HA, i don't know if it is possible, i can help here.
Does the application store something not in the database?
The easyest thing to reach this could be using rqlite as database, like lxd is doing with dqlite or incus with cowsql.
Otherwise i can do it waiting for mariadb support and than there is https://www.dolthub.com/ that can be scalable and replicated. it is like a pain doing so.
the idea is to have like 3 hosts in or not in a incus cluster, replicate the database and installing 3 lxconsole one in each host.

Really thank you for your work!

**Comment by calculatetech** [comment]
I'd like to see SAML rather than OpenID, as well as the option to enforce 2FA for local accounts.

**Comment by pazos** [comment]
Hey, long term user here! Thank you for this awesome software, I use it weekly ;)

I would love to be able to upload and attach isos to virtual machines using lxconsole.
